plugins/inputs/cloudwatch/cloudwatch.go

```diff
@@ -434,7 +434,8 @@ func (c *CloudWatch) fetchNamespaceMetrics() ([]types.Metric, error) {
 			}
 
 			metrics = append(metrics, resp.Metrics...)
-			if resp.NextToken == nil {
+			if resp.NextToken == nil || len(*resp.NextToken) == 0 {
```

Comment:
Is `*resp.NextToken` a string? If so I think
```suggestion
			if resp.NextToken == nil || *resp.NextToken == "" {
```
is more clear.